server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -718,6 +718,74 @@ public Map<String, UploadedSegmentMetadata> getSegmentsUploadedToRemoteStore() {
         return Collections.unmodifiableMap(this.segmentsUploadedToRemoteStore);
     }
 
+    private Map<String, String> populatePreviousMetadata(
+        List<String> sortedMetadataFiles,
+        Set<String> lockedMdFiles,
+        int lastMetadataFilesToKeep
+    ) {
+        Map<String, String> activePreviousMetadata = new HashMap<>();
+        if (sortedMetadataFiles.size() == 0) {
+            return activePreviousMetadata;
+        }
+        String currMetadata = sortedMetadataFiles.get(lastMetadataFilesToKeep);
+        String prevMetadata = null;
+        if (lastMetadataFilesToKeep > 0) {
+            prevMetadata = sortedMetadataFiles.get(lastMetadataFilesToKeep - 1);
+        }
+        activePreviousMetadata.put(currMetadata, prevMetadata);
+        prevMetadata = currMetadata;
+        for (int i = lastMetadataFilesToKeep + 1; i < sortedMetadataFiles.size(); i++) {
+            currMetadata = sortedMetadataFiles.get(i);
+            if (lockedMdFiles.contains(prevMetadata)) {
+                activePreviousMetadata.put(currMetadata, prevMetadata);
+            } else {
+                activePreviousMetadata.put(currMetadata, activePreviousMetadata.get(prevMetadata));
+            }
+            prevMetadata = currMetadata;
+        }
+        return activePreviousMetadata;
+    }
+
+    private Map<String, String> populateNextMetadata(
+        List<String> sortedMetadataFiles,
+        Set<String> lockedMdFiles,
+        int lastMetadataFilesToKeep
+    ) {
+        Map<String, String> activeNextMetadata = new HashMap<>();
+        if (sortedMetadataFiles.size() == 0) {
+            return activeNextMetadata;
+        }
+        activeNextMetadata.put(sortedMetadataFiles.get(sortedMetadataFiles.size() - 1), null);
+        String nextMetadata = sortedMetadataFiles.get(sortedMetadataFiles.size() - 1);
+        for (int i = sortedMetadataFiles.size() - 2; i >= lastMetadataFilesToKeep; i--) {
+            String currMetadata = sortedMetadataFiles.get(i);
+            if (lockedMdFiles.contains(nextMetadata)) {
+                activeNextMetadata.put(currMetadata, nextMetadata);
+            } else {
+                activeNextMetadata.put(currMetadata, activeNextMetadata.get(nextMetadata));
+            }
+            nextMetadata = currMetadata;
+        }
+        return activeNextMetadata;
+    }
+
+    private void processActiveSegments(
```

Comment:
lets take list of metadata files (size always 2) as as an input , instead of calling it twice.